nixos/modules/services/web-servers/garage.nix

```diff
@@ -75,7 +75,17 @@ in
       source = configFile;
     };
 
-    environment.systemPackages = [ cfg.package ]; # For administration
+    environment.systemPackages = if cfg.environmentFile != null
+      then [(
+        # reads the env vars from `cfg.environmentFile` to the garage environment
+        # makes working with the cli quite a bit nicer
+        pkgs.writeScriptBin "garage" ''
+          env - $(cat ${cfg.environmentFile}) ${lib.getExe cfg.package} $@
```

Comment:
I think `$@` should be quoted (`"$@"`) or else it mishandles (splits) arguments containing whitespace.